src/engine.ml

```diff
@@ -1332,9 +1332,40 @@ let wrap_hmac_control now ts mtu session tls_auth key transport outs =
   in
   (session, transport, List.rev outs)
 
-let wrap_tls_crypt_control now ts mtu session tls_crypt key transport outs =
+let wrap_tls_crypt_control now ts mtu session (tls_crypt, wkc) needs_wkc key
+    transport outs =
   let now_ts = ptime_to_ts_exn now in
-  let outs = split_control mtu outs in
+  (* If we reply with hmac cookie we must split such that the first control
+     packet, the Control_wkc has room for the /cleartext/ wkc, and fix the
+     packet length afterwards *)
+  let session, transport, maybe_out, outs =
+    match (needs_wkc, outs) with
+    | true, (`Control, data) :: rest ->
```

Comment:
This is brittle. We need to ensure somehow that we don't send an `ACK` for the server's `HARD_RESET_SERVER_V2` as that the server will not know how to decrypt the ACK.

Reply:
I documented in the relevant places (hopefully) that we should not send out dedicated ACKs for this reason.